src/impls.rs

```diff
@@ -177,6 +181,9 @@ unsafe impl<T, U> Restructure<&ManuallyDrop<T>> for U {
     type Restructured = ManuallyDrop<U>;
 
     unsafe fn restructure(ptr: *mut Self) -> Self::Restructured {
-        unsafe { ::core::ptr::read(ptr.cast()) }
+        #[allow(clippy::undocumented_unsafe_blocks)]
```

Comment:
This block needs to be justified, it should be fixed now if you rebase.

Reply:
ok, then i should rebase now